src/libs/actions/IOU.js

```diff
@@ -75,6 +77,30 @@ function requestMoney(report, amount, currency, recipientEmail, participant, com
         iouReport = ReportUtils.buildOptimisticIOUReport(recipientEmail, debtorEmail, amount, chatReport.reportID, currency, preferredLocale);
     }
 
+    const optimisticTransaction = TransactionUtils.buildOptimisticTransaction(amount * 100, currency, iouReport.reportID, comment);
+    const optimisticTransactionData = {
+        onyxMethod: Onyx.METHOD.MERGE,
```

Comment:
I do think we should use set for the new transaction, AFAIK its how we do this for new reports too right?
```suggestion
        onyxMethod: Onyx.METHOD.SET,
```

Reply:
I think we usually prefer MERGE over SET for performance reasons? Rule of thumb is to use MERGE whenever possible.
https://expensify.slack.com/archives/C01GTK53T8Q/p1682938682704609

Reply:
hmmm, ok, thanks for the link. I thoughts its bettter when its a new occurance of the object

Reply:
Updated, though I think it's a bit of a code smell that it should matter at all